addons/account/models/account_move.py

```diff
@@ -3074,8 +3074,9 @@ def generate_qr_code(self):
         """
         self.ensure_one()
 
-        if not self.is_invoice():
-            raise UserError(_("QR-codes can only be generated for invoice entries."))
+        if self.move_type != 'out_invoice':
```

Comment:
```suggestion
        if self.move_type not in ('out_invoice', 'out_receipt'):
```

Reply:
Or, given you other comment
```suggestion
        if not self.display_qr_code:
```

Reply:
@william-andre @qdp-odoo thank you for your feedbacks, I updated the commits